src/extension.ts

```diff
@@ -965,27 +974,22 @@ function parseName(line) {
     return line.split(' ')[0];
 }
 
-function findPod(callback) {
+function findPod(callback: (pod: PodMetadata) => void) {
     let editor = vscode.window.activeTextEditor;
-    if (!editor) {
-        vscode.window.showErrorMessage('No active editor!');
-        return null; // No open text editor
-    }
-
-    let text = editor.document.getText();
-    try {
-        let obj = yaml.safeLoad(text);
-        if (obj.kind !== 'Pod') {
-            return;
+    if (editor) {
+        let text = editor.document.getText();
+        try {
+            let obj = yaml.safeLoad(text);
+            if (obj.kind && obj.kind === 'Pod' && obj.metadata) {
```

Comment:
perhaps we can move this condition into its own function that returns a boolean so we can do something like :

```
if (isPod(obj)) { 
// etc.
}
```

Reply:
Good call.  In my dreams we would get fancy and declare a `Pod` type and `isPod` would return `obj is Pod` for glorious type safety but life is short.